syntax = "proto3";
package Xianwei;
option cc_generic_services = true;


message UserInfo {
    string user_id = 1;//用户ID
    string nickname = 2;//昵称
    string email = 3; //绑定手机号
}
enum ServerMessageType {
    UserRegisterReqType = 0;
    UserLoginReqType = 1;
    EmailLoginReqType = 2;
    EmailVcodeReqType = 3;
    GetUserInfoReqType = 4;
    SetNicknameReqType = 5;
    SetEmailReqType = 6;
    SetPassword = 7;
    EmailFriendAddReqType = 8; 
    NicknameFriendAddReqType = 9;
    GetFriendApplyReqtype = 10;
    GetFriendListReqType = 11;
    SovelFriendApplyReqType = 12;
}

enum ClientMessageType{
    UserRegisterRspType = 0;
    UserLoginRspType = 1;
    EmailLoginRspType = 2;
    EmailVcodeRspType = 3;
    GetUserInfoRspType = 4;
    SetNicknameRspType = 5;
    SetEmailRspType = 6;
    SetPasswordRspType = 7;
    FriendLoginNoticeType = 8;
    EmailFriendAddRspType = 9;
    NicknameFriendAddRspType = 10;
    GetFriendApplyRsptype = 11;
    GetFriendListRspType = 12;
    SovelFriendApplyRspType = 13;
}

//----------------------------
//用户名注册   
message UserRegisterReq {
    string nickname = 1;
    string password = 2;
    string email = 3;
    string verify_code_id = 4; 
    string verify_code = 5;
}
message UserRegisterRsp {
    bool success = 1;
    string errmsg = 2;
}

enum MessageType {
    file = 0;
    string = 1;
}

message OffMessageofFriend{
    string friend_name = 1;
    MessageType message_type = 2;
    string body = 3;
}

message OffMessageofSession{
    string session_name = 1;
    MessageType message_type = 2;
    string body = 3;
}

//用户名登录 
message UserLoginReq {
    string nickname = 1;
    string password = 2;
}
message UserLoginRsp {
    bool success = 1;
    string errmsg = 2;
    string user_id = 3;
    string email = 4;
    repeated OffMessageofFriend friend = 5;
    repeated OffMessageofSession session = 6;
}



message EmailVerifyCodeReq {
    string email = 1;
}
message EmailVerifyCodeRsp {
    bool success = 1;
    string errmsg = 2;
    string verify_code_id = 3;
    string verify_code  = 4;
}
//----------------------------

//----------------------------

message EmailLoginReq {
    string email = 1;
    string verify_code_id = 2;
    string verify_code = 3;
}
message EmailLoginRsp {
    bool success = 1;
    string errmsg = 2;
    string user_id = 3;
    string email = 4;
    repeated OffMessageofFriend friend = 5;
    repeated OffMessageofSession session = 6;
}

message GetUserInfoReq {
    string user_id = 1;    // 这个字段是网关进行身份鉴权之后填入的字段
}
message GetUserInfoRsp {
    bool success = 1;
    string errmsg = 2; 
    UserInfo user_info = 3;
}
//----------------------------
//用户昵称修改 
message SetUserNicknameReq {
    string user_id = 1;
    string nickname = 2;
}
message SetUserNicknameRsp {
    bool success = 1;
    string errmsg = 2; 
}


message SetUserEmailReq {
    string user_id = 1;
    string email = 2;
    string email_verify_code_id = 3;
    string email_verify_code = 4;
}
message SetUserEmailRsp {
    bool success = 1;
    string errmsg = 2; 
}

message SetUserPasswordReq {
    optional string user_id = 1;
    string password = 2;
    string email_verify_code_id = 3;
    string email_verify_code = 4;
}
message SetUserPasswordRsp{
    bool success = 1;
    string errmsg = 2; 
}

message FriendLoginNotice{
    string name = 1;
}

message EmailFriendAddReq{
    string user_id = 1;
    string email = 2;
}
message EmailFriendAddRsp{
    bool success = 1;
    string errmsg = 2;
}

message NicknameFriendAddReq{
    string user_id = 1;
    string nickname = 2;
}
message NicknameFriendAddRsp{
    bool success = 1;
    string errmsg = 2;
}

message GetFriendApplyReq{
    string user_id = 1;
}
message GetFriendApplyRsp{
    bool success = 1;
    string errmsg = 2;
    repeated UserInfo user_info = 3;
}

message GetFriendListReq{
    string user_id = 1;
}
message GetFriendListRsp{
    bool success = 1;
    string errmsg = 2;
    repeated UserInfo friend_list = 3;
}

message SovelFriendApplyReq{
    string user_id = 1;
    string peer_id = 2;
    bool agree = 3;
}
message SovelFriendApplyRsp{
    bool success = 1;
    string errmsg = 2;
}

message ServerMessage{
    ServerMessageType type = 1;
    oneof payload {
        UserRegisterReq user_register_req = 2;
        UserLoginReq user_login_req = 3;
        EmailLoginReq email_login_req = 4;
        EmailVerifyCodeReq email_verify_code_req = 5;
        GetUserInfoReq get_user_info_req = 6;
        SetUserNicknameReq set_user_nickname_req = 7;
        SetUserEmailReq set_user_email_req = 8;
        SetUserPasswordReq set_user_password_req = 9;
        EmailFriendAddReq email_friend_add_req = 10;
        NicknameFriendAddReq nickname_friend_add_req = 11;
        GetFriendApplyReq get_friend_apply = 12;
        GetFriendListReq get_friend_list_req = 13;
        SovelFriendApplyReq sovel_friend_apply_req = 14;
    }
}

message ClientMessage{
    ClientMessageType type = 1;
    oneof payload {
        UserRegisterRsp user_register_rsp = 2;
        UserLoginRsp user_login_rsp = 3;
        EmailLoginRsp email_login_rsp = 4;
        EmailVerifyCodeRsp email_verify_code_rsp = 5;
        GetUserInfoRsp get_user_info_rsp = 6;
        SetUserNicknameRsp set_user_nickname_rsp = 7;
        SetUserEmailRsp set_user_email_rsp = 8;
        SetUserPasswordRsp set_user_password_rsp = 9;
        FriendLoginNotice friend_login_notice = 10;
        EmailFriendAddRsp email_friend_add_rsp = 11;
        NicknameFriendAddRsp nickname_friend_add_rsp = 12;
        GetFriendApplyRsp get_friend_apply_rsp = 13;
        GetFriendListRsp get_friend_list_rsp = 14;
        SovelFriendApplyRsp sovel_friend_apply_rsp = 15;
    }
}